processing-service-core/api/controllers/tenant.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+var httpHelper = require('../helpers/httpPromiseHelper.js');
+var tenantModel = require('../models/tenant.js');
+
+module.exports = {
+    setTenantConfig: setTenantConfig,
+    setTenantConfigs: setTenantConfigs,
+    deleteTenantConfig: deleteTenantConfig,
+    deleteTenantConfigs: deleteTenantConfigs,
+    getEffectiveTenantConfig: getEffectiveTenantConfig,
+    getEffectiveTenantConfigs: getEffectiveTenantConfigs,
+    getTenantConfig: getTenantConfig,
+    getTenantConfigs: getTenantConfigs
+};
+
+function setTenantConfig(req, response, next) {
+    var createPromise = tenantModel.setTenantConfig(req.swagger.params.tenantId.value, req.swagger.params.key.value,
+            req.swagger.params.value.value);
+    httpHelper.writeCreatePromiseJSONResultToResponse(createPromise, response);
```

Comment:
The contract states this can return either 201 or 204 for create/update. There needs to be some logic here to differentiate between the response result. e.g. have response return '{created: true}' and key off that value.